Testscripts/Linux/VERIFY-LIS-MODULES-VERSION.sh

```diff
@@ -107,44 +119,43 @@ for module in "${HYPERV_MODULES[@]}"; do
 
     # Check to see if the module is loaded
     if [[ $load_status =~ $module ]]; then
-        if [ "$rpmAvailable" = true ] ; then
-            if [ ! -z "$isLISInstalled" ]; then
-                version=$(modinfo "$module" | grep version: | head -1 | awk '{print $2}')
-                LogMsg "$module module: ${version}"
-                if [ "$module" == "mlx4_en" ] ;then
-                    if [ "$MLNX_VERSION" != "$version" ] ;then
-                        LogErr "ERROR: Status: $module $version doesnot match with build version $MLNX_VERSION"
-                        exit_code="1"
-                    fi
-                    continue
-                fi
-                if [ "$expected_lis_version" != "$version" ] ;then
-                    LogErr "ERROR: Status: $module $version doesnot match with build version $expected_lis_version"
-                    exit_code="1"
-                fi
+        LogMsg "LIS module $module loaded successfully"
+        if [ "$rpmAvailable" = true ] && [ ! -z "$isLISInstalled" ]; then
+            version=$(modinfo "$module" | grep version: | head -1 | awk '{print $2}')
+            LogMsg "$module module version: ${version}"
+            if [ "$module" == "mlx4_en" && "$MLNX_VERSION" != "$version" ] ;then
+                    LogErr "Status: $module $version did not match to the build one, $MLNX_VERSION"
+                    exit_code=$((exit_code+1))
+            else
+                continue
+            fi
+            if [ "$expected_lis_version" != "$version" ] ;then
+                LogErr "Status: $module $version did not match to the build one, $expected_lis_version"
+                exit_code=$((exit_code+1))
+            else
                 continue
             fi
         fi
-        
+
         version=$(modinfo "$module" | grep vermagic: | awk '{print $2}')
         if [[ "$version" == "$(uname -r)" ]]; then
-            LogMsg "Found a kernel matching version for $module module: ${version}"
+            LogMsg "Found the matching kernel version of $module module: ${version}"
         else
-            LogErr "Error: LIS module $module doesn't match the kernel build version!"
-            exit_code="1"
+            LogErr "LIS module $module did not match the kernel build version!"
+            exit_code=$((exit_code+1))
         fi
     else
-         LogErr "Error: LIS module $module is not loaded"
-         exit_code="1"
+         LogErr "LIS module $module was not loaded"
+         exit_code=$((exit_code+1))
     fi
 done
 
-if [ "1" -eq "$exit_code" ]; then
-    LogMsg "Exiting with state: TestAborted."
-    SetTestStateAborted
-else
-    LogMsg "Exiting with state: TestCompleted."
+if [ 0 -eq $exit_code ]; then
+    LogMsg "Exiting with state: $__LIS_TESTCOMPLETED."
     SetTestStateCompleted
-fi
-
-exit 0
+    exit 0
+else
+    LogMsg "Exiting with state: $__LIS_TESTABORTED."
+    SetTestStateAborted
+    exit 1
```

Comment:
if the invoked place not use -ignoreLinuxExitCode, and when it exit with 1, we will enter into a long/dead loop.

Reply:
I don't understand the possible case. I see https://github.com/LIS/LISAv2/blob/master/Libraries/CommonFunctions.psm1#L1769 handles the passed and aborted case. because it set SetTestStateAborted and exit with 1.  This script is called in many places directly, but don't see any possible abnormal case. Can you show me possible scenario?

Reply:
@LiliDeng ping here again. Let's clarify this and close ASAP.